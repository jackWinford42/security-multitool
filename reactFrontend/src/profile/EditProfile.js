import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import RamtApi from "../common/Api";
import { Card, CardBody} from 'reactstrap';
import { useHistory } from "react-router-dom";

export default function EditProfile() {
  const [formData, setFormData] = useState({
    username: "",
    profile_pic: ""
  });
  const dispatch = useDispatch();
  const user = useSelector(st => st.currUser);
  console.log(user)
  const history = useHistory();

  async function handleSubmit(evt) {
    evt.preventDefault();
    const res = await RamtApi.editCurrUser(formData)
    console.log("CHECKPOINT")
    console.log(res);
    res.user.email = user.email;
    dispatch({type: "BEGIN_AUTH_SESSION", user: res.user})
    history.push("/profile");
  }

  // Update form data to reflect change in form fields
  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData(data => ({ ...data, [name]: value }));
  }

  return (
    <div className="EditProfile">
      <div id="title">
        <h1>Your Profile</h1>
      </div>
      <div id="profile_content">
        <Card>
          <CardBody>
            <p>Edit details for your account.</p>
            <form className="form-inline" onSubmit={handleSubmit}>
              <label>Username:</label>
              <input
                className="form-control form-control-md"
                name="username"
                placeholder="username"
                value={formData.username}
                onChange={handleChange}
              />
              <label>Profile Picture Url:</label>
              <input
                className="form-control form-control-md"
                name="profile_pic"
                placeholder="profile_pic"
                value={formData.profile_pic}
                onChange={handleChange}
              />
              <button type="submit" className="authButton btn btn-lg btn-primary">
                Edit
              </button>
            </form>
          </CardBody>
        </Card>
      </div>
    </div>
  )
}