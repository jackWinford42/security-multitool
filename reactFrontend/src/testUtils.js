import React from "react";
import { Provider, useDispatch } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducers/rootReducer";
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(thunk)));

const demoUser = {
  email: "jackwinford@gmail.com",
  username: "captainjack",
  password: "qwerty",
  photo_url: "https://bit.ly/3C2VUgQ",
};

// const dispatch = useDispatch();
// await dispatch({type: "BEGIN_AUTH_SESSION", user: demoUser});

const UserProvider =(
    ({children}) =>
    (<Provider store={store}>
      {children}
    </Provider>)
);

export { UserProvider };