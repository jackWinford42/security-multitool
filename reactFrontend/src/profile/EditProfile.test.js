import React from "react";
import { render } from "@testing-library/react";
import EditProfile from "./EditProfile";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils"

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <EditProfile/>
      </UserProvider>
    </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <EditProfile/>
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});