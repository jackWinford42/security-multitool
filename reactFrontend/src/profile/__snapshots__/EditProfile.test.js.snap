// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="EditProfile"
  >
    <div
      id="title"
    >
      <h1>
        Your Profile
      </h1>
    </div>
    <div
      id="profile_content"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <p>
            Edit details for your account.
          </p>
          <form
            class="form-inline"
          >
            <label>
              Username:
            </label>
            <input
              class="form-control form-control-md"
              name="username"
              placeholder="username"
              value=""
            />
            <label>
              Profile Picture Url:
            </label>
            <input
              class="form-control form-control-md"
              name="profile_pic"
              placeholder="profile_pic"
              value=""
            />
            <button
              class="authButton btn btn-lg btn-primary"
              type="submit"
            >
              Edit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
